travisci/check_patch.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+difference=$(git show origin/master..@ | clang-format-diff-3.9 -p 1 -style=file);
```

Comment:
Are we enforcing clang format on all files? If not we can do something for example to prune for ones we care about:
$CLANG_FORMAT --version
files_to_lint="$(git diff --name-only --diff-filter=ACMRTUXB $TRAVIS_COMMIT_RANGE | grep '^src/[^.]*[.]\(cpp\|h\)$' || true)"
for f in $files_to_lint; do
    d=$(diff -u "$f" <($CLANG_FORMAT "$f") || true)
    if ! [ -z "$d" ]; then
        echo "!!! $f not compliant to coding style, here is the fix:"
        echo "$d"
        fail=1
    fi
done

Reply:
hmm, that is what clang-format-diff does :)

Reply:
Ahh, it looks like its cut off there ha, but I was referring to the right side where it prunes cpp and h files " grep '^src/[^.]*[.]\(cpp\|h\)$'"

Reply:
We have mostly cpp and h files in the repo with a few sh/readme files. IIRC clang only formats cpp and h files?

Reply:
Yeah, we don't need it. It's just to avoid it checking Android.mk and Makefiles since it'll try to format those too. But unless someone changes a file other than the cpp, h files I don't see too much of a big issue.

Reply:
Looks like clang format diff already ignores makefiles. It uses this regex:
[Clang-format-diff regex](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/clang-format-diff.py#L47)